src/plugins/explorer/sync-transactions.js

```diff
@@ -221,7 +239,7 @@ function createSyncer (config, eventBus, web3, queue, eventsRegistry, indexer) {
       .then(() =>
         Promise.all([
           getPastCoinTransactions(0, bestBlock, address),
-          getPastEvents(0, bestBlock, address)
+          getPastEvents(0, bestBlock, address, function (syncedBlock) { bestBlock = syncedBlock })
```

Comment:
The on-progress callback should be received in `refreshAllTransactions` and passed over to `getPastEvents`. Client code should also make sure the synced block is persisted.

Reply:
I thought the same, but I kept the functionality as it was - currently refreshing provided no onProgress functionality (not sure why). I will review it